src/Point/Point.cs

```diff
@@ -81,15 +86,116 @@ public async Task SimulateTrailed()
+
+            if (_currentConnection != null)
+            {
+                ReportedDegradedPointPosition? reportedDegradedPointPosition = _hasNoNonCrucialPointMachines ?
+                    ReportedDegradedPointPosition.DegradedPointPositionIsNotApplicable :
+                    GetReportedDegradedPointPosition(_pointState.LatestFinalPointPosition);
+
+                if (reportedDegradedPointPosition != null)
+                {
+                    ReportedPointPosition? reportedPointPosition = GetReportedPointPositionDegraded(message.Position);
+                    if (reportedPointPosition != null)
+                    {
+                        UpdatePointState((ReportedPointPosition)reportedPointPosition, (ReportedDegradedPointPosition)reportedDegradedPointPosition);
+                        var occupancyStatus = new PointPointPositionMessage(_localId, _remoteId, _pointState.PointPosition, _pointState.DegradedPointPosition);
+                        await _currentConnection.RequestStream.WriteAsync(new SciPacket() { Message = ByteString.CopyFrom(occupancyStatus.ToByteArray()) });
+                    }
+                }
+            }
+
+            await UpdateConnectedWebClients();
+        }
+
+        /// <summary>
+        /// Sets the <c>ReportedPointPosition</c> to the latest final position (left/right).
+        /// </summary>
+        /// <returns></returns>
+        public async Task FinalizePosition(){
+            
+            if (_currentConnection != null)
+            {
+                if (_pointState.PointPosition != ReportedPointPosition.PointIsInARightHandPositionDefinedEndPosition &&
+                _pointState.PointPosition != ReportedPointPosition.PointIsInALeftHandPositionDefinedEndPosition) {
+                    ReportedDegradedPointPosition reportedDegradedPointPosition = _hasNoNonCrucialPointMachines ?
+                            ReportedDegradedPointPosition.DegradedPointPositionIsNotApplicable : ReportedDegradedPointPosition.PointIsNotInADegradedPosition;
+                    
+                    UpdatePointState(_pointState.LatestFinalPointPosition, reportedDegradedPointPosition);
+
+                    var occupancyStatus = new PointPointPositionMessage(_localId, _remoteId, _pointState.PointPosition, _pointState.DegradedPointPosition);
+                    await _currentConnection.RequestStream.WriteAsync(new SciPacket() { Message = ByteString.CopyFrom(occupancyStatus.ToByteArray()) });
+                }
+            }
+            await UpdateConnectedWebClients();
+        }
+
         protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
+            _hasNoNonCrucialPointMachines = _configuration.GetSection("PointSettings").Get<PointConfiguration>()?.hasNoNonCrucialPointMachines ?? false;
+            var tmp_hasNoNonCrucialPointMachines = _configuration["crucial-point-machines"];

```

Comment:
Okay, so the intended effect of using the configuation record to make this code look nicer didn't work out... :smile:. Let's fix this.